core/src/main/java/com/datastax/oss/driver/api/core/type/codec/TypeCodec.java

```diff
@@ -82,19 +82,39 @@ default boolean accepts(@NonNull GenericType<?> javaType) {
   }
 
   /**
-   * Whether this codec is capable of processing the given Java type.
+   * Whether this codec is capable of processing the given Java class.
    *
-   * <p>The default implementation wraps the class in a generic type and calls {@link
-   * #accepts(GenericType)}, therefore it is invariant as well.
+   * <p>This implementation simply compares the given class (or its wrapper type if it is a
+   * primitive type) against this codec's runtime (raw) class; it is <em>invariant</em> with respect
+   * to the passed argument (through the usage of {@link Class#equals(Object)} and <em>it's strongly
+   * recommended not to modify this behavior</em>. This means that a codec will only ever return
+   * {@code true} for the <em>exact</em> runtime (raw) Java class that it has been created for.
    *
    * <p>Implementors are encouraged to override this method if there is a more efficient way. In
-   * particular, if the codec targets a non-generic class, the check can be done with {@code
-   * Class#isAssignableFrom}. Or even better, with {@code ==} if that class also happens to be
-   * final.
+   * particular, if the codec targets a final class, the check can be done with a simple {@code ==}.
    */
   default boolean accepts(@NonNull Class<?> javaClass) {
     Preconditions.checkNotNull(javaClass);
-    return accepts(GenericType.of(javaClass));
+    if (javaClass.isPrimitive()) {
```

Comment:
That makes me think, we should probably also test for primitive classes in implementations, for example in `IntCodec`:
```java
  @Override
  public boolean accepts(@NonNull Class<?> javaClass) {
    return javaClass == Integer.class || javaClass == int.class;
  }
```

Reply:
Indeed, I will push the changes in a separate commit.

Reply:
To be clear: codecs always store the wrapper types for primitive types, so in practice there should be no lookup in the registry by primitive types, but it doesn't hurt to add that.